pkg/asset/installconfig/openstack/validation/machinepool.go

```diff
@@ -119,8 +119,13 @@ func validateFlavor(flavor Flavor, req flavorRequirements, fldPath *field.Path)
 	if flavor.VCPUs < req.VCPUs {
 		errs = append(errs, fmt.Sprintf("Must have minimum of %d VCPUs, had %d", req.VCPUs, flavor.VCPUs))
 	}
-	if flavor.Disk < req.Disk {
-		errs = append(errs, fmt.Sprintf("Must have minimum of %d GB Disk, had %d GB", req.Disk, flavor.Disk))
+	// Here we tolerate the minimum storage requirements to come from
+	// Disk or Ephemeral.
+	// The root disk is an ephemeral disk that the base image is copied into
+	// whereas an ephemeral disk offers to the machine a local disk storage
+	// linked to the lifecycle of a VM instance.
```

Comment:
trhough explanation, but I don't think we need to leave the comment in the code.  The code below is pretty clear, and adding the note just creates the risk of the comment becoming stale. Anyone changing the code can easily forget to edit the comment.

Reply:
Fair enough, I'll follow your judgement and remove the comment.